cmd/cli/job/list_test.go

```diff
@@ -0,0 +1,120 @@
+//go:build integration || !unit
+
+package job_test
+
+import (
+	"encoding/csv"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	cmdtesting "github.com/bacalhau-project/bacalhau/cmd/testing"
+	"github.com/bacalhau-project/bacalhau/pkg/docker"
+	"github.com/bacalhau-project/bacalhau/pkg/system"
+)
+
+// In order for 'go test' to run this suite, we need to create
+// a normal test function and pass our suite to suite.Run
+func TestListSuite(t *testing.T) {
+	suite.Run(t, new(ListSuite))
+}
+
+// Define the suite, and absorb the built-in basic suite
+// functionality from testify - including a T() method which
+// returns the current testing context
+type ListSuite struct {
+	cmdtesting.BaseSuite
+}
+
+// Before each test
+func (s *ListSuite) SetupTest() {
+	docker.MustHaveDocker(s.T())
+	s.BaseSuite.SetupTest()
+}
+
+func (s *ListSuite) TestJobList() {
+	// Submit a job
+	args := []string{
+		"docker", "run",
+		"--wait",
+		"alpine", "echo", "hello world",
+	}
+	_, out, err := s.ExecuteTestCobraCommand(args...)
+	require.NoError(s.T(), err, "Error submitting job")
+	jobID := system.FindJobIDInTestOutput(out)
+
+	// Run 'bacalhau job list'
+	_, listOut, err := s.ExecuteTestCobraCommand("job", "list")
+	require.NoError(s.T(), err, "Error running job list")
+
+	// Check that the shortened job ID appears in the list
+	require.Contains(s.T(), listOut, jobID[:10], "Short Job ID not found in list output")
+}
+
+func (s *ListSuite) TestJobListLimit() {
+	// Submit multiple jobs
+	numJobs := 5
+	jobIDs := make([]string, numJobs)
+	for i := 0; i < numJobs; i++ {
+		args := []string{
+			"docker", "run",
+			"--wait",
+			"alpine", "echo", fmt.Sprintf("hello world %d", i),
+		}
+		_, out, err := s.ExecuteTestCobraCommand(args...)
+		require.NoError(s.T(), err, "Error submitting job")
+		jobID := system.FindJobIDInTestOutput(out)
+		jobIDs[i] = jobID
+	}
+
+	// Run 'bacalhau job list --limit 2 --output csv'
+	_, listOut, err := s.ExecuteTestCobraCommand("job", "list", "--wide", "--limit", "2", "--output", "csv")
+	require.NoError(s.T(), err, "Error running job list")
```

Comment:
Any particular reason you chose to use CSV output here instead of JSON? 

JSON will be easier to read, and you do not need to worry about CSV headers, etc

Reply:
No particular reason. Switched to JSON